### DISCUSSION TITLE
Resource class throw 'Undefined property' error by intelephense.

### DISCUSSION THREAD

**Discussion by Neozips-Benjamin** [laravel/framework#45511] [Q&A]
After updating php to 8.2, resource class throw 'Undefined property' error.
How can I fix it?

I can disable intelephense 
<img width="652" alt="image" src="https://user-images.githubusercontent.com/112033890/210655093-0ed21a60-b54c-4242-8790-df0b4693830b.png">

Or I can put #[AllowDynamicProperties] top of Resource Class.

However, I would like to know the property way to write the code with php 8.2 standard.
'maintenances' table has 'id' and 'request_id' columns.

**Resource Class**
```php
class MaintenanceResource extends JsonResource
{
    public function toArray($request)
    {
        return [
            'id' => $this->id,
            'request_id' => $this->request_id,
        ];
    }
```

**Model Class**
```php
class Maintenance extends Model
{
    use HasFactory;
    protected $table = 'maintenances';
    protected $guarded = [];
}
```

**Comment by egofinder** [comment]
After I update larval 9.47.0, php.8.2.1, php intelephense 1.9.3 resource class doesn't throw 'Undefined property' error!!
